core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/solver/scope/SolverScope.java

```diff
@@ -194,6 +197,10 @@ public void setBestSolutionTimeMillis(Long bestSolutionTimeMillis) {
         this.bestSolutionTimeMillis = bestSolutionTimeMillis;
     }
 
+    public SupplyManager getPillarSupplyManager() {
+        return pillarSupplyManager;
+    }
```

Comment:
I don't like that everything ends up on `SolverScope`.
But how else can I share the pillar supply manager across all possible pillar moves?

Reply:
getSolverScope().getScoreDirector().getSupplyManager().demand(....)

There should only be 1 SupplyManager instance per ScoreDirector, and it's not pillar specific.
The demand and the supply are pillar specific.